src/components/task-plan/reading.cjsx

```diff
@@ -82,91 +99,99 @@ SelectTopics = React.createClass
+    id = @getPlanId()
+    TaskPlanActions.updateDueAt(id, value)
 
   setTitle: ->
+    id = @getPlanId()
     value = @refs.title.getDOMNode().value
-    TaskPlanActions.updateTitle(@state.id, value)
+    TaskPlanActions.updateTitle(id, value)
 
-  publish: ->
-    {id} = TaskPlanStore.get(@state.id)
+  publishPlan: ->
+    id = @getPlanId()
     TaskPlanActions.publish(id)
     @transitionTo('editReading', {id})
 
-  renderTopics: (topicId) ->
-    topic = TocStore.getSectionInfo(topicId)
-    <li>{topic.number} - {topic.title}</li>
+  deletePlan: () ->
+    id = @getPlanId()
+    @transitionTo('dashboard')
+    TaskPlanActions.delete(id)
+    {}
 
   render: ->
-    id = @getParams().id
-    id = @state.id unless id
+    id = @getPlanId()
 
-    if TaskPlanStore.isLoaded(id)
-
-      plan = TaskPlanStore.get(id)
-
-      isEnabled = plan.title and plan.due_at and plan.settings.page_ids.length > 0
-
-      classes = []
-      classes.push('disabled') unless isEnabled
-      classes = classes.join(' ')
+    unless TaskPlanStore.isLoaded(id)
+      <div className="loading">Loading...</div>
 
-      footer = <BS.Button bsStyle="primary" className={classes} onClick={@publish}>Publish</BS.Button>
-      headerText = if id then 'Edit Reading' else 'Add Reading'
-      dueDate = if @state.due_at then @state.due_at else @props.due_at
-      topics = TaskPlanStore.getTopics(@state.id)
+    plan = TaskPlanStore.get(id)
 
+    isEnabled = plan?.title and plan?.due_at and plan?.settings?.page_ids?.length > 0
 
-      selectedReadingList =
-        <ul className="selected-reading-list">
-          <li><strong>Currently selected sections in this reading</strong></li>
-          {_.map(topics, @renderTopics)}
-        </ul> if topics?.length
-
-      <BS.Panel bsStyle="default" className="create-reading" footer={footer}>
-        <h1>{headerText}</h1>
-        <div>
-          <label htmlFor="title">Name</label>
-          <input ref="title" id="title" type="text" onChange={@setTitle} value={@props.title}/>
-        </div>
-        <div>
-          <label htmlFor="due-date">Due Date</label>
-          <DateTimePicker
-            id="due-date"
-            format="MMM dd, yyyy"
-            time={false}
-            calendar={true}
-            readOnly={false}
-            onChange={@setDueAt}
-            value={dueDate}/>
-        </div>
-        <p>
-          <label>Select Readings</label>
-          <SelectTopics planId={@state.id} selected={topics}/>
-        </p>
-        {selectedReadingList}
-      </BS.Panel>
+    headerText = if id then 'Edit Reading' else 'Add Reading'
+    topics = TaskPlanStore.getTopics(id)
+    
+    footer= <ReadingFooter enabled={isEnabled} onPublish={@publishPlan} onDelete={@deletePlan}/>
 
-    else
-      <div className="loading">Loading...</div>
+    <BS.Panel bsStyle="default" className="create-reading" footer={footer}>
+      <h1>{headerText}</h1>
+      <div>
+        <label htmlFor="title">Name</label>
```

Comment:
Do you mean the label for?  I just added that in there for clarity a while back, I don't really think it's necessary.  I'll take it out.